icu4c/source/tools/icuexportdata/icuexportdata.cpp

```diff
@@ -293,3 +245,143 @@ int main(int argc, char* argv[]) {
 
     return 0;
 }
+
+#define INCLUDED_FROM_UCASE_CPP
+#include "ucase_props_data.h"
+
+struct AddRangeHelper {
+    UMutableCPTrie* ucptrie;
+};
+
+static UBool U_CALLCONV
+addRangeToUCPTrie(const void* context, UChar32 start, UChar32 end, uint32_t value) {
+    IcuToolErrorCode status("addRangeToUCPTrie");
+    UMutableCPTrie* ucptrie = ((const AddRangeHelper*) context)->ucptrie;
+    umutablecptrie_setRange(ucptrie, start, end, value, status);
+    handleError(status, "setRange");
+
+    return TRUE;
+}
+
+int exportCase() {
+    if (options[OPT_INDEX].doesOccur) {
```

Comment:
I can see it being helpful to be warned if you try using an option that has no effect. On the other hand, I can also imagine that it could be a maintenance burden to keep adding all these checks in various places as we add more modes. I'm fine either way; if you prefer to fail gracefully, I'm happy to remove these checks.